File: PAMO/app/src/main/java/pl/pamo/ui/bmrcalculator/BmrCalculator.java
package pl.pamo.ui.bmrcalculator;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import java.text.DecimalFormat;

import pl.pamo.R;

public class BmrCalculator extends Fragment {

    /*
      Dla mężczyzn przyjmuje postać:
        66,5 + 13,75M + 5,003W – 6,775 L

      Dla kobiet natomiast:

        655,1 + (9,563*M) + (1,85*W) – (4,676 *L)
       gdzie:

       M – oznacza masę ciała w kilogramach
       W – wzrost w centymetrach
       L – wiek w latach


       */
    private static final DecimalFormat decimalFormat = new DecimalFormat("#.##");
    private TextView scoreTextView;
    private EditText inputWeight;
    private EditText inputHeight;
    private EditText inputAge;
    private String gender;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_harris_benedict_calculator, container, false);

        inputWeight = rootView.findViewById(R.id.inputWeightEditText);

        inputHeight = rootView.findViewById(R.id.inputHeightEditText);

        inputAge = rootView.findViewById(R.id.inputAgeEditText);

        scoreTextView = rootView.findViewById(R.id.scoreView);

        Spinner genderSpinner = rootView.findViewById(R.id.spinner);
        Button countBtn = rootView.findViewById(R.id.countBtn);

        ArrayAdapter<CharSequence> staticAdapter = ArrayAdapter.createFromResource(
                getContext(), R.array.gender_list, android.R.layout.simple_spinner_item);

        staticAdapter.setDropDownViewResource(android.R.layout.simple_spinner_item);
        genderSpinner.setAdapter(staticAdapter);

        genderSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                gender = (String) parent.getItemAtPosition(position);

            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {
                gender = "Kobieta";
            }
        });

        countBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                printBmrScore(countBMR());
            }


        });

        return rootView;
    }


    private void printBmrScore(double score) {
        scoreTextView.setText(String.format("%s%s", getString(R.string.score_bmr_message), decimalFormat.format(score)));
    }


    private double countBMR() {
        double weight = Double.parseDouble(inputWeight.getText().toString());
        double height = Double.parseDouble(inputHeight.getText().toString());
        double age = Double.parseDouble(inputAge.getText().toString());
        double score;
        switch (gender) {

            case "Male":
                score = 66.5 + (13.75 * weight) + (5.003 * height) - (6.775 * age);
                break;
            case "Female":
                score = 655.1 + (9.563 * weight) + (1.85 * height) - (4.676 * age);
                break;
            default:
                throw new IllegalStateException("Unexpected value: " + gender);
        }
        return score;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
    }
}

File: PAMO/app/src/main/java/ui/bmrcalculator/BmrCalculator.kt
package ui.bmrcalculator

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import android.widget.AdapterView.OnItemSelectedListener
import androidx.fragment.app.Fragment
import pl.pamo.R
import java.text.DecimalFormat

class BmrCalculator : Fragment() {
    private var scoreTextView: TextView? = null
    private var inputWeight: EditText? = null
    private var inputHeight: EditText? = null
    private var inputAge: EditText? = null
    private var gender: String? = null
    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val rootView = inflater.inflate(R.layout.fragment_harris_benedict_calculator, container, false)
        inputWeight = rootView.findViewById(R.id.inputWeightEditText)
        inputHeight = rootView.findViewById(R.id.inputHeightEditText)
        inputAge = rootView.findViewById(R.id.inputAgeEditText)
        scoreTextView = rootView.findViewById(R.id.scoreView)
        val genderSpinner = rootView.findViewById<Spinner>(R.id.spinner)
        val countBtn = rootView.findViewById<Button>(R.id.countBtn)
        val staticAdapter = ArrayAdapter.createFromResource(
                requireContext(), R.array.gender_list, android.R.layout.simple_spinner_item)
        staticAdapter.setDropDownViewResource(android.R.layout.simple_spinner_item)
        genderSpinner.adapter = staticAdapter
        genderSpinner.onItemSelectedListener = object : OnItemSelectedListener {
            override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
                gender = parent.getItemAtPosition(position) as String
            }

            override fun onNothingSelected(parent: AdapterView<*>?) {
                gender = "Kobieta"
            }
        }
        countBtn.setOnClickListener { printBmrScore(countBMR()) }
        return rootView
    }

    private fun printBmrScore(score: Double) {
        scoreTextView!!.text = String.format("%s%s", getString(R.string.score_bmr_message), decimalFormat.format(score))
    }

    private fun countBMR(): Double {
        val weight = inputWeight!!.text.toString().toDouble()
        val height = inputHeight!!.text.toString().toDouble()
        val age = inputAge!!.text.toString().toDouble()
        val score: Double
        score = when (gender) {
            "Male" -> 66.5 + 13.75 * weight + 5.003 * height - 6.775 * age
            "Female" -> 655.1 + 9.563 * weight + 1.85 * height - 4.676 * age
            else -> throw IllegalStateException("Unexpected value: $gender")
        }
        return score
    }

    override fun onDestroyView() {
        super.onDestroyView()
    }

    companion object {
        /*
      Dla mężczyzn przyjmuje postać:
        66,5 + 13,75M + 5,003W – 6,775 L

      Dla kobiet natomiast:

        655,1 + (9,563*M) + (1,85*W) – (4,676 *L)
       gdzie:

       M – oznacza masę ciała w kilogramach
       W – wzrost w centymetrach
       L – wiek w latach


       */
        private val decimalFormat = DecimalFormat("#.##")
    }
}